server/pkg/controller/stripe.go

```diff
@@ -409,6 +408,70 @@ func (c *StripeController) handleInvoicePaid(event stripe.Event) (ente.StripeEve
 	return ente.StripeEventLog{UserID: userID, StripeSubscription: *stripeSubscription, Event: event}, nil
 }
 
+func (c *StripeController) handlePaymentIntentFailed(event stripe.Event, country ente.StripeAccountCountry) (ente.StripeEventLog, error) {
+	var paymentIntent stripe.PaymentIntent
+	json.Unmarshal(event.Data.Raw, &paymentIntent)
+	// Figure out the user
+	invoiceID := paymentIntent.Invoice.ID
+	client := c.StripeClients[country]
+	invoice, err := client.Invoices.Get(invoiceID, nil)
+	if err != nil {
+		return ente.StripeEventLog{}, stacktrace.Propagate(err, "")
+	}
+	// void the invoice, in case the payment intent failed
```

Comment:
can't void the invoice as it leads to subscription going into a active state, instead of expired / payment_due